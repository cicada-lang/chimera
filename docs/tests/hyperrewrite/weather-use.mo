hyperrule rain {
  [Rain(), Car()] => [Rain(), InCar()]
  [Rain()] => [WearUmbrella()]
}

hyperrule sun {
  [Sun()] => [WearSunGlasses(), Cool()]
}

hyperrule weather {
  [SunRain()] => [Sun(), Rain()]
  use rain
  use sun
}

print weather(quote [SunRain()])
print weather(quote [SunRain(), Car()])
