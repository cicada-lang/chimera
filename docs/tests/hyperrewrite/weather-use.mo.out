[WearUmbrella(), WearSunGlasses(), Cool()]
[InCar(), WearUmbrella(), WearSunGlasses(), Cool()]
